src/examples/Stroke.cpp

```diff
@@ -89,13 +89,21 @@ void tvgDrawCmds(tvg::Canvas* canvas)
 
     //Test for Stroke Width
     for (int i = 0; i < 10; ++i) {
-        auto shape = tvg::Shape::gen();
-        shape->moveTo(50, 550 + (25 * i));
-        shape->lineTo(750, 550 + (25 * i));
-        shape->stroke(255, 255, 255, 255);       //color: r, g, b, a
-        shape->stroke(i + 1);                    //stroke width
-        shape->stroke(tvg::StrokeCap::Round);    //default is Square
-        if (canvas->push(move(shape)) != tvg::Result::Success) return;
+        auto hline = tvg::Shape::gen();
+        hline->moveTo(50, 550 + (25 * i));
+        hline->lineTo(400, 550 + (25 * i));
+        hline->stroke(255, 255, 255, 255);       //color: r, g, b, a
+        hline->stroke(i + 1);                    //stroke width
+        hline->stroke(tvg::StrokeCap::Round);    //default is Square
+        if (canvas->push(move(hline)) != tvg::Result::Success) return;
+
+        auto vline = tvg::Shape::gen();
+        vline->moveTo(450 + (33 * i), 550);
+        vline->lineTo(450 + (33 * i), 780);
+        vline->stroke(255, 255, 255, 255);       //color: r, g, b, a
+        vline->stroke(i + 1);                    //stroke width
+        vline->stroke(tvg::StrokeCap::Round);    //default is Square
+        if (canvas->push(move(vline)) != tvg::Result::Success) return;
     }
```

Comment:
@hermet for the completeness I'd add the code below, but it's your call:

```
    auto line1 = tvg::Shape::gen();
    line1->moveTo(320, 580);
    line1->lineTo(490, 580);
    line1->stroke(255, 255, 255, 255);       //color: r, g, b, a
    line1->stroke(15);                      
    line1->stroke(tvg::StrokeCap::Round);

    auto line2 = std::unique_ptr<tvg::Shape>((tvg::Shape*)line1->duplicate());
    line2->stroke(tvg::StrokeCap::Square);
    line2->translate(0, 50);

    auto line3 = std::unique_ptr<tvg::Shape>((tvg::Shape*)line1->duplicate());
    line3->stroke(tvg::StrokeCap::Butt);
    line3->translate(0, 100);

    if (canvas->push(move(line1)) != tvg::Result::Success) return;
    if (canvas->push(move(line2)) != tvg::Result::Success) return;
    if (canvas->push(move(line3)) != tvg::Result::Success) return;
```
result:

Reply:
Make a patch!